spec/features/events/event_availability_spec.rb

```diff
@@ -0,0 +1,97 @@
+          start_time: @event.start_time - 1.hour, end_time: @event.end_time)
+
+        # This person is available, staying late
+        av_responder_arrive_ontime_leave_late = create(:person, department: @mrc)
+        create(:availability, person: av_responder_arrive_ontime_leave_late, status: 'Available',
+          start_time: @event.start_time, end_time: @event.end_time + 1.hour)
+
+        av_responder_arrive_early_leave_late = create(:person, department: @mrc)
+        create(:availability, person: av_responder_arrive_early_leave_late, status: 'Available',
+          start_time: @event.start_time - 1.hour, end_time: @event.end_time + 1.hour)
+
+        # These people should be partially available
+        pav_responder_arrive_late_leave_ontime = create(:person, department: @mrc)
+        create(:availability, person: pav_responder_arrive_late_leave_ontime, status: 'Available',
+          start_time: @event.start_time + 1.hour, end_time: @event.end_time)
+
+        pav_responder_arrive_late_leave_late = create(:person, department: @mrc)
+        create(:availability, person: pav_responder_arrive_late_leave_late, status: 'Available',
+          start_time: @event.start_time + 1.hour, end_time: @event.end_time + 1.hour)
+
+        pav_responder_arrive_late_leave_early = create(:person, department: @mrc)
+        create(:availability, person: pav_responder_arrive_late_leave_early, status: 'Available',
+          start_time: @event.start_time + 1.hour, end_time: @event.end_time - 1.hour)
+
+        pav_responder_arrive_ontime_leave_early = create(:person, department: @mrc)
+        create(:availability, person: pav_responder_arrive_ontime_leave_early, status: 'Available',
+          start_time: @event.start_time, end_time: @event.end_time - 1.hour)
+
+        uav_responder_exact_times = create(:person, department: @mrc)
+        create(:availability, person: uav_responder_exact_times,
+          status: 'Unavailable',
+          start_time: @event.start_time, end_time: @event.end_time - 1.hour)
+
+        uav_responder_exact_start_late_end = create(:person, department: @mrc)
+        create(:availability, person: uav_responder_exact_start_late_end,
+          status: 'Unavailable',
+          start_time: @event.start_time, end_time: @event.end_time + 1.hour)
+
+        uav_responder_early_start_exact_end = create(:person, department: @mrc)
+        create(:availability, person: uav_responder_early_start_exact_end,
+          status: 'Unavailable',
+          start_time: @event.start_time - 1.hour, end_time: @event.end_time)
+
+        uav_responder_early_start_late_end = create(:person, department: @mrc)
+        create(:availability, person: uav_responder_early_start_late_end,
+          status: 'Unavailable',
+          start_time: @event.start_time - 1.hour, end_time: @event.end_time + 1.hour)
+
+        visit event_path(@event)
+save_and_open_page
```

Comment:
please remove save_and_open_page.... unless you want it to open the page every time. in which case leave it. :)